File: mediq-frontend/src/services/reportService.js
import api from './api';

export const reportService = {
  getCampAnalytics: async (campId) => api.get(`/reports/camp/${campId}`),
  getDoctorWorkloadReport: async (campId) => api.get(`/reports/doctors/${campId}`),
  getMedicineConsumptionReport: async () => api.get('/reports/medicines'),
  getReferralReport: async (campId) => api.get(`/reports/referrals/${campId}`),
  globalSearch: async (query) => api.get('/search', { params: { query } }),
  getPatientsCsvUrl: (campId) => `/api/v1/reports/export/patients/csv${campId ? `?campId=${campId}` : ''}`,
  getMedicinesCsvUrl: () => '/api/v1/reports/export/medicines/csv',
  downloadPatientsCsv: async (campId) => api.get(`/reports/export/patients/csv${campId ? `?campId=${campId}` : ''}`, { responseType: 'blob' }),
  downloadMedicinesCsv: async () => api.get('/reports/export/medicines/csv', { responseType: 'blob' }),
};

export default reportService;

File: mediq-backend/src/main/java/com/mediq/service/ReportService.java
package com.mediq.service;

import com.mediq.dto.*;

import java.util.List;

public interface ReportService {

    CampAnalyticsResponse getCampAnalytics(Long campId);

    List<DoctorReportResponse> getDoctorWorkloadReport(Long campId);

    List<MedicineReportResponse> getMedicineConsumptionReport();

    ReferralReportResponse getReferralReport(Long campId);

    GlobalSearchResultResponse globalSearch(String query);
}
